### PR TITLE
Redesign sidebar menu UI

### LINKED ISSUE #27168: Improving sidebar menu UI
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

The sidebar menu in dashboard always has scrollbar even all sub menus are closed (screen height < 1080px)

Screenshot 2022-01-02 at 18-36-17 Dashboard • develop

### Describe the solution you'd like

The category title can be redesigned to decrease the height of sidebar menu.

Screenshot 2022-01-02 at 18-36-09 Dashboard • sidebar

### Alternatives you've considered

_No response_

### Additional context

_No response_

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by mparvazi**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Change padding and margin values to reduce the height of sidebar. Redesign category title. Unifies the UI between legacy and new theme. Fix scrollbar background on hover for new theme.
| Type?             | improvement
| Category?         | BO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27168
| How to test?      | Go to BO > Dashboard for legacy theme. Go to BO > Orders for new theme.
| Possible impacts? | The sidebar menu UI in BO.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27271)

## Legacy theme (desktop)
Screenshot 2022-01-07 at 16-11-06 Dashboard

## New theme (desktop)
Screenshot 2022-01-07 at 16-11-18 Orders

### CODE REVIEW

**Comment by mparvazi** [comment]
I used the color of submenu-arrow for category-title.
Code: `#6c868e`

I think it looks like the sample designed by @prestascott



Background color of sidebar menu is: `#363a41`
Variations for this color:
Screenshot from 2022-01-07 16-37-48
Colors in middle can be used for category-title: `#757E8C` or `#858C99`

| `#6c868e`  | `#757E8C` | `#858C99` |
| ------------- | ------------- | ------------- |
| 6c868e  | 757E8C | 858C99 |

**Review by matthieu-rolland — COMMENTED** [review]
thank you @mparvazi 👍 

looks good to me, let's wait for @NeOMakinG 's opinion ;)

File: admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss [hunk 1]
```diff
@@ -71,7 +72,7 @@
       display: flex;
       align-items: center;
       padding: 0.875rem;
-      border-bottom: 1px solid $white;
+      border-bottom: 1px solid #bebebe;
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss]:
```suggestion
      border-bottom: 1px solid $gray-dark-text;
```

Please use a variable

File: admin-dev/themes/default/scss/partials/_nav.scss [hunk 2]
```diff
@@ -448,24 +448,18 @@
...
     font-size: 1rem;
-    font-weight: 500;
-    line-height: 1.25rem;
-    border-top: none;
   }
 
   & > .title {
-    color: #fff;
-    background: $gray-dark-menu;
+    color: #6c868e;
```
Comment (NeOMakinG) [inline admin-dev/themes/default/scss/partials/_nav.scss]:
```suggestion
    color: $gray-500;
```

**Review by NeOMakinG — DISMISSED** [review]
👍

File: admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss [hunk 3]
```diff
@@ -476,22 +486,18 @@
...
+    margin: 0.75rem 0 0 1rem;
     font-size: 1rem;
-    font-weight: 500;
-    border-top: none;
   }
 
   & > .title {
-    color: $white;
-    background: $gray-dark;
+    color: #6c868e;
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss]: Oh wow sorry forgot this one and I've another problem

Comment (mparvazi) [reply admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss]: It should be $gray-medium
I'm testing...

File: admin-dev/themes/default/scss/partials/_nav.scss [hunk 4]
```diff
@@ -448,24 +448,18 @@
...
     font-size: 1rem;
-    font-weight: 500;
-    line-height: 1.25rem;
-    border-top: none;
   }
 
   & > .title {
-    color: #fff;
-    background: $gray-dark-menu;
+    color: $gray-500;
```
Comment (NeOMakinG) [inline admin-dev/themes/default/scss/partials/_nav.scss]:
```suggestion
    color: $medium-gray;
```

Comment (NeOMakinG) [reply admin-dev/themes/default/scss/partials/_nav.scss]: Didn't see it was the default theme on this one, sorry :)

**Review by NeOMakinG — DISMISSED** [review]
👍

**Review by atomiix — DISMISSED** [review]

**Comment by prestascott** [comment]
Hey everyone 👋

Thanks for your PR @mparvazi,

I've checked the contrast ratio between your color proposition with the background color `#363A41`, and unfortunately none of them passes the recommended contrast ratio (at least 4:5:1 in Accessibility).

I used this [Contrast checker tool](https://webaim.org/resources/contrastchecker/) and here are the results:

For `#6C868E` - [Link](https://webaim.org/resources/contrastchecker/?fcolor=858C99&bcolor=363A41)
image 6 (1)

For `#757E8C` - [Link](https://webaim.org/resources/contrastchecker/?fcolor=858C99&bcolor=363A41)
image 4 (1)

For `#858C99` - [Link](https://webaim.org/resources/contrastchecker/?fcolor=858C99&bcolor=363A41)
image 7 (1)

So what I did is to generate a palette with this [Material Design Palette Generator tool](http://mcg.mbitson.com/#!?mcgpalette0=%236c868e) using the secondary color `#6C868E` as the 500-tone. 
I picked up the color `#98AAB0` which is the 300-tone and checked the contrast ratio.

Here is the result ([Link](https://webaim.org/resources/contrastchecker/?fcolor=858C99&bcolor=363A41)):
image 5

The contrast ratio is 4.73:1 which is above the recommended 4:5:1.

Also, I noticed that the color used for text in the sidebar menu wasn't white, so to increase the contrast ratio and improve the Accessibility of the interface, I definitely suggest using the white color `#FFFFFF` for text.

Here is the final design:
Navigation drawer

Wdyt?

**Comment by mparvazi** [comment]
@prestascott 

`#98AAB0` is very good for category-title.

But for links color:
`#ffffff` is used for active link and hover.
So if we use `#ffffff` for links, what color we should use for active link and hover?

Current style (left column in below table):
Link color: `#bebebe`
Active link color: `#ffffff`

Orders link is selected:

| Link color: `#bebebe`  | Link color: `#ffffff` |
| ------------- | ------------- |
|   |  | 

I used `#98AAB0` for category title in screenshots.

**My suggestion**:

link-color: `#d3dbdd` (based on [this](http://mcg.mbitson.com/#!?mcgpalette0=%236c868e) color palette)
active-link-color: `#ffffff`
category-title-color: `#98aab0`
sub-menu-arrow-color: `#98aab0`
menu-collapse-icon-color: `#98aab0`

**Comment by prestascott** [comment]
Your link-color seems a little too bright and close to the white color, here is what I suggest:
- For category-title: `#98AAB0`
- For sub-menu-arrow-color: `#FFFFFF`
- For link-levelone: `#FFFFFF`
- For link-leveltwo: `#B6C3C7` which is the 200-tone, and it still passes the contrast ratio with 6:32:1 ([Link](https://webaim.org/resources/contrastchecker/?fcolor=B6C3C7&bcolor=363A41))
- For active link-leveltwo: `#FFFFFF`

Also, based on this [Color palette](http://mcg.mbitson.com/#!?mcgpalette0=%23363a41) we can use `#22252A` (800-tone) for active background.

Wdyt?

**Comment by mparvazi** [comment]
@prestascott 

category-title-color: `#98AAB0`
link-levelone-color: `#FFFFFF`
link-leveltwo-color: `#B6C3C7`
active-link-leveltwo-color: `#FFFFFF`
link-active-background: `#22252A`
link-levelone-hover-background: `#22252A`
submenu-background: `#22252A`
submenu-arrow-color: `#FFFFFF`
menu-collapse-icon-color: `#FFFFFF`

**Comment by prestascott** [comment]
Perfect, we're all good!

**Comment by mparvazi** [comment]
@prestascott 

Changing collapse-menu icon:

| Open  | Collapsed |
| ------------- | ------------- |
|   |  |

Wdyt?

If you're interested in it, then we can try with different icons to select the best:
https://jossef.github.io/material-design-icons-iconfont/

**Comment by mparvazi** [comment]
@NeOMakinG 

New variables:
admin-dev/themes/new-theme/scss/config/_settings.scss
```
// Sidebar Menu
$sidebar-background-color: $dark-gray;
$sidebar-active-background-color: #22252A;
$sidebar-text-color: #98AAB0; 
$sidebar-primary-link-color: $white; 
$sidebar-secondary-link-color: #B6C3C7; 
$sidebar-active-icon-color: $blue;
```
Wdyt?

**Comment by prestascott** [comment]
@mparvazi 

Interesting proposition but it's not really intuitive. The burger menu icon is mainly used when there is a lack of space to design. For example in smaller devices such as mobile.

When clicking on the burger menu icon, it should expand the sidebar menu. On your proposition, it's the opposite.

I think we should keep it simple and keep the keyboard double arrow left and right. It's easier to understand that it expands and collapses the sidebar.

Wdyt?

**Comment by mparvazi** [comment]
@prestascott 

I've got your point.
Keep it simple :ok_hand:

**Comment by mparvazi** [comment]
### Dashboard (desktop)
Screenshot 2022-01-17 at 14-19-39 Dashboard • sidebar

### Dashboard (mobile)


### Orders (desktop)
Screenshot 2022-01-17 at 14-19-59 Orders • sidebar

### Orders (mobile)

**Comment by NeOMakinG** [comment]
Where is the ux-only close icon in the menu on migrated pages? Is it related to your PR?

File: admin-dev/themes/default/scss/modules/_variables.scss [hunk 5]
```diff
@@ -169,14 +169,18 @@ $paddingLeftMobile: 0.625rem;
 $widthSidebarSubmenu: 200px;
 $menu-item-size: 34px;
 $min-height: 950px;
-$gray-dark-text: #bebebe;
-$gray-dark-text-hover: #fff;
+$gray-dark-link: #fff;
+$gray-dark-link-hover: #fff;
```
Comment (NeOMakinG) [inline admin-dev/themes/default/scss/modules/_variables.scss:173]: Normally I would say use the $white variable, but if we wan'ts to change the $white variable I think we don't want menu links to be #000 but a grayer color so it's fine to me!

File: admin-dev/themes/default/scss/partials/_nav.scss [hunk 6]
```diff
@@ -485,7 +497,7 @@
   }
 
   .material-icons {
-    color: #bebebe;
+    color: #fff;
```
Comment (NeOMakinG) [inline admin-dev/themes/default/scss/partials/_nav.scss]:
```suggestion
    color: $gray-dark-link;
```

I think we want this one to be the same color as links no?

Comment (NeOMakinG) [reply admin-dev/themes/default/scss/partials/_nav.scss]: Anyway, we need to use an SCSS variable

**Comment by mparvazi** [comment]
> Where is the ux-only close icon in the menu on migrated pages? Is it related to your PR?

I didn't change it.

**Review by NeOMakinG — DISMISSED** [review]

**Comment by mparvazi** [comment]
Adding space before first item in submenu:

**Review by NeOMakinG — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @mparvazi 

Thanks for your suggestion. The improvement is well done!

**Desktop (legacy theme) :** 
image

**Mobile (legacy theme) :** 
image

**Desktop (New theme) :** 
image

**Mobile (New theme) :** 
image

category-title-color: #98AAB0 ==> OK
link-levelone-color: #FFFFFF ==> OK
link-leveltwo-color: #B6C3C7 ==> OK
active-link-leveltwo-color: #FFFFFF ==> OK
link-active-background: #22252A ==> OK
link-levelone-hover-background: #22252A ==> OK
submenu-background: #22252A ==> OK
submenu-arrow-color: #FFFFFF ==> OK
menu-collapse-icon-color: #FFFFFF ==> OK

image
**background-color** 
image
image
image
image

**firefox :** 
https://watch.screencastify.com/v/1e0NgNud5VyRV0G2H0jZ

**responsive mode :** 
https://watch.screencastify.com/v/DyJTSg37J0QdXkpwmn34

Also, i checked with multistore and different languages (As RTL, Spanish, China etc) : OK
image

So, QA :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @mparvazi & @HanaRebaiQA

**Pull request merged**